src/sentry/static/sentry/app/views/performance/compare/utils.tsx

```diff
@@ -0,0 +1,772 @@
+
+    // the best candidate span is one that has the closest start timestamp to baselineSpan;
+    // and one that has a duration that's close to baselineSpan
+
+    const baselineSpanDuration = Math.abs(
+      baselineSpan.timestamp - baselineSpan.start_timestamp
+    );
+
+    const {regressionSpan, index} = candidates.reduce((bestCandidate, nextCandidate) => {
+      const {regressionSpan: thisSpan} = bestCandidate;
+      const {regressionSpan: otherSpan} = nextCandidate;
+
+      // calculate the deltas of the start timestamps relative to baselineSpan's
+      // start timestamp
+
+      const deltaStartTimestampThisSpan = Math.abs(
+        thisSpan.start_timestamp - baselineSpan.start_timestamp
+      );
+
+      const deltaStartTimestampOtherSpan = Math.abs(
+        otherSpan.start_timestamp - baselineSpan.start_timestamp
+      );
+
+      // calculate the deltas of the durations relative to the baselineSpan's
+      // duration
+
+      const thisSpanDuration = Math.abs(thisSpan.timestamp - thisSpan.start_timestamp);
+      const otherSpanDuration = Math.abs(otherSpan.timestamp - otherSpan.start_timestamp);
+
+      const deltaDurationThisSpan = Math.abs(thisSpanDuration - baselineSpanDuration);
+      const deltaDurationOtherSpan = Math.abs(otherSpanDuration - baselineSpanDuration);
+
+      const thisSpanScore = deltaDurationThisSpan + deltaStartTimestampThisSpan;
+      const otherSpanScore = deltaDurationOtherSpan + deltaStartTimestampOtherSpan;
+
+      if (thisSpanScore < otherSpanScore) {
+        // sort thisSpan before otherSpan
+        return bestCandidate;
+      }
+
+      if (thisSpanScore > otherSpanScore) {
+        // sort thisSpan after otherSpan
+        return nextCandidate;
+      }
+
+      return bestCandidate;
+    });
+
+    // remove regressionSpan from list of remainingRegressionChildren
+    remainingRegressionChildren.splice(index, 1);
+
+    comparablePairs.push({
+      type: 'descendent',
+      parent_span_id,
+      baselineSpan,
+      regressionSpan,
+    });
+
+    children.push({
+      comparisonResult: 'matched',
+      span_id: generateMergedSpanId({baselineSpan, regressionSpan}),
+      op: baselineSpan.op,
+      description: baselineSpan.description,
+      baselineSpan,
+      regressionSpan,
+    });
+  }
+
+  // push any remaining un-matched regressionSpans
+  for (const regressionSpan of remainingRegressionChildren) {
+    children.push({
+      comparisonResult: 'regression',
+      regressionSpan,
+    });
+  }
+
+  // sort children by most time added
+
+  children.sort(sortByMostTimeAdded);
+
+  return {
+    comparablePairs,
+    children,
+  };
+}
+
+function matchableSpans({
+  baselineSpan,
+  regressionSpan,
+}: {
+  baselineSpan: SpanType;
+  regressionSpan: SpanType;
+}): boolean {
+  const opNamesEqual = baselineSpan.op === regressionSpan.op;
+  const descriptionsEqual = baselineSpan.description === regressionSpan.description;
+
+  return opNamesEqual && descriptionsEqual;
```

Comment:
For spans that contain SQL queries this is going to miss quite a bit. Should we look at other characteristics like location in the tree?

Reply:
This is a naive comparison right now.

I've considered making use of edit distance or levenshtein distance as a heuristic to know that span descriptions are similar. I wanted to wait until feedback before I try that approach. 

Tree location is already considered when match making spans.

Reply:
It looks like SQL queries get matched up fairly well if they're less specific: 

<img width="1668" alt="Screen Shot 2020-07-30 at 4 58 19 PM" src="https://user-images.githubusercontent.com/139499/88974031-fe784680-d285-11ea-8223-27f3ba44d8e3.png">